pkg/controller/storagecluster/storagecluster.go

```diff
@@ -192,9 +230,50 @@ func (c *Controller) syncStorageCluster(
 		return nil
 	}
 
-	// TODO: Handle history of specs for multiple upgrades
-	// Cleanup history once all pods are updated and using the same spec
+	// Set defaults in the storage cluster object if not set
+	if err := c.setStorageClusterDefaults(cluster); err != nil {
+		return fmt.Errorf("failed to update StorageCluster %v/%v with default values: %v",
+			cluster.Namespace, cluster.Name, err)
+	}
 
+	// Construct histories of the StorageCluster, and get the hash of current history
+	cur, old, err := c.constructHistory(cluster)
+	if err != nil {
+		return fmt.Errorf("failed to construct revisions of StorageCluster %v/%v: %v",
+			cluster.Namespace, cluster.Name, err)
+	}
+	hash := cur.Labels[defaultStorageClusterUniqueLabelKey]
+
+	// TODO: Don't process a storage clsuter until all its previous creations and
+	// deletions have been processed.
+	err = c.manage(cluster, hash)
+	if err != nil {
+		return err
+	}
+
+	switch cluster.Spec.UpdateStrategy.Type {
+	case corev1alpha1.OnDeleteStorageClusterStrategyType:
```

Comment:
we should warn that ondelelete isn't supported and that we are falling back on RollingUpdate.

Reply:
Also add an issue to add support for this

Reply:
It is actually supported. Basically we don't update the pods here until it is deleted.